### ISSUE TITLE
Markers sometimes don't get rendered

### ISSUE THREAD

**Issue by 7mllm7** [react-community/react-native-maps#334]
I have this `render()` function. Using `redux`, and upon changing `user.location` and/or `call.location`, I expect the markers to change their coordinates.
However I can see that they don't always update / render.
It looks like an issue, can you concur?

```
const userImage = require('../../images/user.png');
const callImage = require('../../images/call.png');

render() {
    return (
        <MapView
            style={styles.map}
            initialRegion={{
                latitude: -2.2848,
                longitude: 42.7995,
                latitudeDelta: 1,
                longitudeDelta: 1
              }}
        >
          {(() => {
            let user = this.props.user;
            if (user) {
              return <MapView.Marker
                  coordinate={{
                        longitude: user.getIn(['location', 'lon']),
                        latitude: user.getIn(['location', 'lat'])
                      }}
                  title="My Location"
                  image={userImage}
              />
            }
          })()}

          {(() => {
            let call = this.props.call;
            if (call) {
              return <MapView.Marker
                    coordinate={{
                          longitude: call.getIn(['location', 'lon']),
                          latitude: call.getIn(['location', 'lat'])
                        }}
                    title={call.title}
                    description={call.description}
                    image={callImage}
                  />;
            }
          })()}

          {(() => {
            let call = this.props.call;
            if (call) {
              return <MapView.Polyline
                  coordinates={call.getIn(['route', 'steps']).toJS()}
                  strokeColor='#EC1B2E'
                  strokeWidth={2}
              />
            }
          })()}
        </MapView>
    );
  }

...
const styles = StyleSheet.create({
  map: {
    flex: 1,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0
  }
});
```

**Comment by 7mllm7** [comment]
Makes me wonder, what is the best way to update dynamic markers? Looks like re-`render`ing them by updating their coordinates is a little clumsy...

**Comment by 7mllm7** [comment]
UPDATE: after updating to 0.6.2, I see it's either solved or rare on Android, but still happens on iOS.

**Comment by jrichardlai** [comment]
Hello, can you try using a key?

**Comment by 7mllm7** [comment]
Tried, didn't help...

On this matter - since I don't really have a unique id to provide from my model (`call`, or `user` in the example above), should the key for each component be unique in each render (e.g. `key={++someGlobalKey}`), or, since I know I only have 3 components, can I just hardcode it to `key={0}`, `key={1}`, `key={2}`?

Thanks!

**Comment by carleryd** [comment]
I'm having the same problem. Whenever I change my markers array some of the markers don't get rendered, either the ones that were already present, or new ones that should appear.
I always get the correct result if I first set the markers array to be empty(markers = []) just before I set it to be new values, because then it renders a map without any markers, and then the new ones.
I tried setting all new markers to unique key values but this did not help.
